scripts/zones/Cape_Teriggan/Zone.lua

```diff
@@ -4,56 +4,59 @@
 --
 -----------------------------------
 local ID = require("scripts/zones/Cape_Teriggan/IDs")
-require("scripts/globals/icanheararainbow");
-require("scripts/globals/conquest");
-require("scripts/globals/weather");
-require("scripts/globals/zone");
+-----------------------------------
+require("scripts/globals/icanheararainbow")
+require("scripts/globals/conquest")
+require("scripts/globals/weather")
+require("scripts/globals/zone")
 -----------------------------------
 
 function onInitialize(zone)
-    UpdateNMSpawnPoint(ID.mob.KREUTZET);
-    GetMobByID(ID.mob.KREUTZET):setRespawnTime(math.random(900, 10800));
+    UpdateNMSpawnPoint(ID.mob.KREUTZET)
+    GetMobByID(ID.mob.KREUTZET):setRespawnTime(math.random(32400,43200)) -- 9 to 12 hours
+    GetMobByID(ID.mob.KREUTZET):setLocalVar("cooldown",os.time() + mob:getRespawnTime()/1000)
+    DisallowRespawn(GetMobByID(ID.mob.KREUTZET):getID(), true) -- prevents accidental 'pop' during no wind weather and immediate despawn
 
     dsp.conq.setRegionalConquestOverseers(zone:getRegionID())
-end;
+end
 
 function onConquestUpdate(zone, updatetype)
     dsp.conq.onConquestUpdate(zone, updatetype)
-end;
+end
 
 function onZoneIn( player, prevZone)
-    local cs = -1;
+    local cs = -1
 
-    if (player:getXPos() == 0 and player:getYPos() == 0 and player:getZPos() == 0) then
-        player:setPos( 315.644, -1.517, -60.633, 108);
+    if player:getXPos() == 0 and player:getYPos() == 0 and player:getZPos() == 0 then
+        player:setPos( 315.644, -1.517, -60.633, 108)
     end
 
-    if (triggerLightCutscene(player)) then -- Quest: I Can Hear A Rainbow
-        cs = 2;
+    if triggerLightCutscene(player) then -- Quest: I Can Hear A Rainbow
+        cs = 2
     end
 
-    return cs;
-end;
+    return cs
+end
 
 function onRegionEnter( player, region)
-end;
+end
 
 function onEventUpdate( player, csid, option)
-    if (csid == 2) then
-        lightCutsceneUpdate(player); -- Quest: I Can Hear A Rainbow
+    if csid == 2 then
+        lightCutsceneUpdate(player) -- Quest: I Can Hear A Rainbow
     end
-end;
+end
 
 function onEventFinish( player, csid, option)
-    if (csid == 2) then
-        lightCutsceneFinish(player); -- Quest: I Can Hear A Rainbow
+    if csid == 2 then
+        lightCutsceneFinish(player) -- Quest: I Can Hear A Rainbow
     end
-end;
+end
 
 function onZoneWeatherChange(weather)
-    if (GetMobAction(ID.mob.KREUTZET) == 24 and (weather == dsp.weather.WIND or weather == dsp.weather.GALES)) then
-        SpawnMob(ID.mob.KREUTZET); -- Kreutzet
-    elseif (GetMobAction(ID.mob.KREUTZET) == 16 and (weather ~= dsp.weather.WIND and weather ~= dsp.weather.GALES)) then
-        DespawnMob(ID.mob.KREUTZET);
+    if GetMobAction(ID.mob.KREUTZET) == 0 and os.time() > GetMobByID(ID.mob.KREUTZET):getLocalVar("cooldown") and
```

Comment:
we should replace `GetMobAction()` checks (it is deprecated) with `isSpawned()` checks. You'll have to do a `GetMobByID()` to use it though.